Add docs for using Gitlab CI with Docker wormhole executor (#4770)
testcontainers/testcontainers-java: docs/supported_docker_environment/continuous_integration/gitlab_ci.md

# GitLab CI

> ## example with docker.sock

Reviewer: ```suggestion
## Example using Docker socket mounting
```

> If you have your own docker runner installed, and you have `/var/run/docker.sock` mounted in the gitlabrunner.

Reviewer: ```suggestion
This applies if you have your own GitlabCI runner installed, use the Docker executor and you have `/var/run/docker.sock` mounted in the runner configuration.
```

> See below for example configuration for your own docker gitlab runner: 

Reviewer: ```suggestion
See below for an example runner configuration: 
```

```toml
[[runners]]
  name = "MACHINE_NAME"
  url = "https://gitlab.com/"
  token = "GENERATED_GITLAB_RUNNER_TOKEN"
  executor = "docker"
  [runners.docker]
    tls_verify = false
    image = "docker:latest"
    privileged = false
    disable_entrypoint_overwrite = false
    oom_kill_disable = false
    disable_cache = false
    volumes = ["/var/run/docker.sock:/var/run/docker.sock", "/cache"]
    shm_size = 0
```

> Please include the following in the test task for `.gitlab-ci.yml`:

Reviewer: ```suggestion
Please also include the following in your GitlabCI pipeline definitions (`.gitlab-ci.yml`) that use Testcontainers:
```

```yml
variables:
  TESTCONTAINERS_HOST_OVERRIDE: "host.docker.internal"
```

> The environment variable `TESTCONTAINERS_HOST_OVERRIDE` needs to be configured otherwise a wrong ip address would be 
> assigned to resolve the host of the runner, which could lead to failing tests.

Reviewer: ```suggestion
The environment variable `TESTCONTAINERS_HOST_OVERRIDE` needs to be configured, otherwise, a wrong IP address would be used to resolve the Docker host, which will likely lead to failing tests.
```

> ## example with dind - docker-in-docker service

Reviewer: ```suggestion
## Example using DinD (Docker-in-Docker)
```
